api/v1alpha1/databasesecretengineconfig_types.go

```diff
@@ -124,9 +92,8 @@ func (r *DatabaseSecretEngineConfig) setInternalCredentials(context context.Cont
 			log.Error(err, "unable to retrieve RandomSecret", "instance", r)
 			return err
 		}
-		secret, err := vaultClient.Logical().Read(randomSecret.GetPath())
+		secret, err := GetVaultKV2Secret(randomSecret.GetPath(), context)
 		if err != nil {
-			log.Error(err, "unable to retrieve vault secret", "instance", r)
```

Comment:
I like to print error every time event if there is nothing to add...

Reply:
Error is printed by ReadSecret method, it will a duplication of the error if I add it back